File: day01.py
def calculate_fuel(mass):
    return int(mass / 3) - 2

modules = []
with open('inputs/01') as f:
    modules = list(map(lambda l: int(l.rstrip('\n')), f.readlines()))

print('Part 1:', sum(map(calculate_fuel, modules)))

p2 = 0
for mass in modules:
    current_fuel = calculate_fuel(mass)
    total_fuel = current_fuel
    while True:
        current_fuel = calculate_fuel(current_fuel)
        if current_fuel <= 0:
            break
        else:
            total_fuel += current_fuel
    p2 += total_fuel

print('Part 2:', p2)

File: day01.nim
import utils
import sequtils
import strutils
import math

proc calculateFuel(mass: int): int =
    return (mass / 3).floor.toInt - 2

proc part1(input: seq[int]): int =
    return input.map(calculateFuel).sum

proc part2(input: seq[int]): int =
    var total = 0
    for mass in input:
        var currentFuel = calculateFuel(mass)
        var totalFuel = currentFuel
        while true:
            currentFuel = calculateFuel(currentFuel)
            if currentFuel <= 0:
                break
            else:
                totalFuel += currentFuel
        total += totalFuel
    return total


if isMainModule:
    let input = readInputLines("01").map(parseInt)
    echo part1(input)
    echo part2(input)
